core/src/main/java/nu/yona/server/subscriptions/entities/BuddyConnectionChangeMessage.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.subscriptions.entities;
+
+import javax.persistence.Entity;
+import javax.persistence.Transient;
+
+import nu.yona.server.Translator;
+import nu.yona.server.crypto.seckey.SecretKeyUtil;
+import nu.yona.server.messaging.entities.BuddyMessage;
+
+@Entity
+public abstract class BuddyConnectionChangeMessage extends BuddyMessage
```

Comment:
Can we add a unit test that a `BuddyConnectionChangeMessage` works as expected when `firstName` and `lastName` are `null`? So for instance:
- `getFirstName_firstNameNull_returnsAlternativeContainingNickname`
- `encrypt_firstAndLastNameNull_doesNotThrow`
- `decrypt_firstAndLastNameNull_doesNotThrow`

Reply:
Created [YD-558](https://jira.yona.nu/browse/YD-558).